types/node_definition.go

```diff
@@ -65,10 +65,7 @@ func (n *NodeDefinition) GetStartupConfig() string {
 }
 
 func (n *NodeDefinition) GetConfigDispatcher() *ConfigDispatcher {
-	if n == nil {
-		return nil
-	}
-	if n.Config == nil {
+	if n == nil || n.Config == nil {
 		return &ConfigDispatcher{}
 	}
 	return n.Config
```

Comment:
why returning an empty struct here? that implies that a config is present. 
nil means it was not configured

Reply:
yes, that is a dirty solution. I did this because we have the following code piece that merges maps of the `*ConfigDispatcher`
https://github.com/srl-labs/containerlab/blob/fix-nilderef/types/topology.go#L138-L151

in order to avoid nil pointer deref in case where I access Vars field like this `ndef.GetConfigDispatcher().Vars` I decided to return an initialized struct

I can rewrite that merge piece to first get the *ConfigDispatcher, and if it is not nil to try and access its Vars

Reply:
maybe add a `GetConfigDispatcherVars` method ?

Reply:
okay, that sounds fine
like that, I suppose d2f7473?